greenmail-core/src/main/java/com/icegreen/greenmail/imap/commands/SortCommand.java

```diff
@@ -0,0 +1,326 @@
+
+        @Override
+        public int compare(StoredMessage m1, StoredMessage m2) {
+            try {
+                int compareResult = 0;
+                for (SortKey sortKey : sortTerm.sortCriteria) {
+                    switch (sortKey) {
+                        case REVERSE:
+                            reverse.set(true);
+                            break;
+                        case ARRIVAL:
+                            compareResult = doCompare(m1.getReceivedDate(), m2.getReceivedDate());
+                            break;
+                        case CC:
+                            compareResult = doCompare(getCc(m1), getCc(m2));
+                            break;
+                        case DATE:
+                            compareResult = doCompare(m1.getMimeMessage().getSentDate(), m2.getMimeMessage().getSentDate());
+                            break;
+                        case FROM:
+                            compareResult = doCompare(getFrom(m1), getFrom(m2));
+                            break;
+                        case SIZE:
+                            compareResult = doCompare(m1.getMimeMessage().getSize(), m2.getMimeMessage().getSize());
+                            break;
+                        case SUBJECT:
+                            compareResult = doCompare(m1.getMimeMessage().getSubject(), m2.getMimeMessage().getSubject());
+                            break;
+                        case TO:
+                            compareResult = doCompare(getTo(m1), getTo(m2));
+                            break;
+                        default:
+                            break;
+                    }
+                    if (compareResult != 0) {
+                        break;
+                    }
+                }
+                return compareResult;
+            } catch (MessagingException e) {
+                throw new RuntimeException(e.getMessage(), e);
+            }
+        }
+
+        private String getFrom(StoredMessage message) throws MessagingException {
+            Address[] addresses = message.getMimeMessage().getFrom();
+            if (addresses == null || addresses.length == 0) {
+                return null;
+            }
+            return String.valueOf(addresses[0]);
+        }
+
+        private String getTo(StoredMessage message) throws MessagingException {
+            Address[] addresses = message.getMimeMessage().getRecipients(Message.RecipientType.TO);
+            if (addresses == null || addresses.length == 0) {
+                return null;
+            }
+            return String.valueOf(addresses[0]);
+        }
+
+        private String getCc(StoredMessage message) throws MessagingException {
+            Address[] addresses = message.getMimeMessage().getRecipients(Message.RecipientType.CC);
+            if (addresses == null || addresses.length == 0) {
+                return null;
+            }
+            return String.valueOf(addresses[0]);
```

Comment:
Isn't this assuming that the number of recipients in the cc field == 1 address? I don't think you can assume that.